localstack/services/cloudformation/provider.py

```diff
@@ -153,6 +159,37 @@ def create_stack(self, context: RequestContext, request: CreateStackInput) -> Cr
                 )
             state.stacks.pop(existing.stack_id)
 
+        if (
+            "CAPABILITY_AUTO_EXPAND" not in request.get("Capabilities", [])
+            and "Transform" in template
```

Comment:
```
"Transform" in template
```

That seems quite a bit too generic? I'd suggest actually parsing and checking for the corresponding keys